pages/verification/index.tsx

```diff
@@ -0,0 +1,169 @@
+import { Helmet } from "react-helmet";
+import React, { useState } from "react";
+import LayoutContainer from "@components/LayoutContainer";
+import api from "@shared/fetcher";
+import apiAnonymous from "@shared/fetcherAnonymous";
+import { useCheckExchangeTokenMutation } from "@services/auth/authHooks";
+import { useRouter } from "next/router";
+import { GetServerSidePropsContext } from "next";
+import Cookies from "js-cookie";
+import { ITokenEnum, MessageEnum, UserEnum } from "@shared/enum";
+import { useInitChatAnonymousMutation } from "@services/initChatAnonymous/initChatAnonymousHooks";
+import { toast } from 'react-toastify'
+import getConfig from "next/config";
+import { useGenerateAnonUserInfoMutation } from "@services/generateAnonUserInfo/generateAnonUserInfoHooks";
+import { LoaderWrapper } from "@components/LoaderWrapper";
+
+interface PageProps {
+    exchangeToken?: string
+    isSendMessage?: boolean
+    isFailedVerify?: boolean
+}
+
+const { publicRuntimeConfig } = getConfig();
+
+export default function Verification(props: PageProps) {
+    const router = useRouter();
+    const exchangeToken = useCheckExchangeTokenMutation();
+    const generateAnonUser = useGenerateAnonUserInfoMutation();
+    const initChatAnon = useInitChatAnonymousMutation();
+    const [isLoading, setIsLoading] = useState(props.exchangeToken ? true : false)
+    const getLoginPage = () => {
+        fetch('/api/getWebLogin').then(async (res) => {
+            toast('Sending you to humanID for verification', {
+                autoClose: false,
+                type: 'info',
+            })
+            const data = await res.json();
+            if (data.data?.webLoginUrl) {
+                router.push(data.data?.webLoginUrl)
+            }
+        })
+    }
+
+    React.useEffect(() => {
+        const member = localStorage.getItem(MessageEnum.targetUser);
+        const message = localStorage.getItem(MessageEnum.tempMessage);
+        if (!!props.exchangeToken && props.isSendMessage && !props.isFailedVerify) {
+            toast('Receiving anonymous identifier from humanID', {
+                autoClose: 3000,
+                type: 'info',
+                progressStyle: {
+                    background: '#4782D7'
+                },
+            })
+            exchangeToken.mutate({ exchangeToken: props.exchangeToken }, {
+                onSuccess: (data) => {
+                    if (data) {
+                        Cookies.set(ITokenEnum.token, data?.token);
+                        Cookies.set(ITokenEnum.refreshToken, data?.refresh_token);
+                        Cookies.set(ITokenEnum.anonymousToken, data?.anonymousToken);
+                        Cookies.set(UserEnum.humanId, data?.data.human_id);
+                        Cookies.set(UserEnum.userId, data?.data.user_id);
+                        apiAnonymous.defaults.headers.common['Authorization'] = `Bearer ${data.anonymousToken}`;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Storing sensitive information like tokens in cookies can expose your application to Cross-Site Scripting (XSS) attacks. Consider storing them in HttpOnly cookies or in-memory storage like sessionStorage or localStorage. However, please note that each storage method has its own trade-offs in terms of security and usability.

85, 95, 106:
Console errors are great for debugging during development but they should be removed or replaced with proper logging in production code. Consider using a logging library that can provide more context about the error and can be turned off in production builds.



<!-- This is an auto-generated comment by OSS CodeRabbit -->